geowave/src/main/scala/geotrellis/spark/io/geowave/GeowaveAttributeStore.scala

```diff
@@ -0,0 +1,256 @@
+
+import com.vividsolutions.jts.geom._
+import mil.nga.giat.geowave.adapter.raster.adapter.RasterDataAdapter
+import mil.nga.giat.geowave.core.geotime.ingest._
+import mil.nga.giat.geowave.core.geotime.store.statistics.BoundingBoxDataStatistics
+import mil.nga.giat.geowave.core.index.ByteArrayId
+import mil.nga.giat.geowave.core.index.HierarchicalNumericIndexStrategy
+import mil.nga.giat.geowave.core.index.HierarchicalNumericIndexStrategy.SubStrategy
+import mil.nga.giat.geowave.core.store._
+import mil.nga.giat.geowave.core.store.index.{PrimaryIndex, CustomIdIndex}
+import mil.nga.giat.geowave.core.store.operations.remote.options.DataStorePluginOptions
+import mil.nga.giat.geowave.core.store.query.QueryOptions
+import mil.nga.giat.geowave.datastore.accumulo._
+import mil.nga.giat.geowave.datastore.accumulo.index.secondary.AccumuloSecondaryIndexDataStore
+import mil.nga.giat.geowave.datastore.accumulo.metadata._
+import mil.nga.giat.geowave.datastore.accumulo.operations.config.AccumuloRequiredOptions
+import mil.nga.giat.geowave.mapreduce.input.{GeoWaveInputKey, GeoWaveInputFormat}
+import org.apache.accumulo.core.client.{TableNotFoundException, ZooKeeperInstance}
+import org.apache.accumulo.core.client.security.tokens.PasswordToken
+import org.apache.hadoop.mapreduce.Job
+import org.apache.log4j.Logger
+import org.apache.spark.{Logging, SparkConf, SparkContext}
+import org.apache.spark.SparkContext._
+import org.geotools.coverage.grid._
+import org.geotools.gce.geotiff._
+import org.opengis.coverage.grid.GridCoverage
+import org.opengis.parameter.GeneralParameterValue
+
+import scala.collection.JavaConverters._
+import scala.util.Try
+
+import resource._
+import spray.json._
+import spray.json.DefaultJsonProtocol._
+
+
+object GeowaveAttributeStore {
+
+  def accumuloRequiredOptions(
+    zookeepers: String,
+    accumuloInstance: String,
+    accumuloUser: String,
+    accumuloPass: String,
+    geowaveNamespace: String
+  ): AccumuloRequiredOptions = {
+    val aro = new AccumuloRequiredOptions
+    aro.setZookeeper(zookeepers)
+    aro.setInstance(accumuloInstance)
+    aro.setUser(accumuloUser)
+    aro.setPassword(accumuloPass)
+    aro.setGeowaveNamespace(geowaveNamespace)
+    aro
+  }
+
+  def basicOperations(
+    zookeepers: String,
+    accumuloInstance: String,
+    accumuloUser: String,
+    accumuloPass: String,
+    geowaveNamespace: String
+  ): BasicAccumuloOperations = {
+    return new BasicAccumuloOperations(
+      zookeepers,
+      accumuloInstance,
+      accumuloUser,
+      accumuloPass,
+      geowaveNamespace)
+  }
+
+  def adapters(bao: BasicAccumuloOperations): Array[RasterDataAdapter] = {
+    for (adapters <- managed(new AccumuloAdapterStore(bao).getAdapters)) {
+      val list = Iterator.iterate(adapters)({ _ => adapters })
+        .takeWhile({ _ => adapters.hasNext })
+        .map(_.next.asInstanceOf[RasterDataAdapter])
+        .toArray
+
+      return list
+    }
+    Array.empty[RasterDataAdapter]
+  }
```

Comment:
I would lose the return, as it's not idiomatic to return as a control struct.

A possible rewrite of this function:

```scala
def adapters(bao: BasicAccumuloOperations): Array[RasterDataAdapter] =
  for (adapters <- managed(new AccumuloAdapterStore(bao).getAdapters)) {
    adapters.toScala
      .map({ case rda: RasterDataAdapter => rda })
      .toArray
  }
```